tests/cxx/pragma_associated.cc

```diff
@@ -0,0 +1,25 @@
+//===--- pragma_associated.cc - test input file for iwyu ------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "tests/cxx/pragma_associated-d1.h"  // IWYU pragma: associated
```

Comment:
I figured that would be covered by the auto-associated header, but I guess double manual association would be nice to test as well. Will add.